backend/src/Designer/Services/Implementation/ResourceRegistryService.cs

```diff
@@ -308,48 +303,80 @@ public async Task<XacmlPolicy> GetXacmlPolicy(string serviceCode, int serviceEdi
             getAccessListsResponse.EnsureSuccessStatusCode();
-            AccessList accessList = await getAccessListsResponse.Content.ReadAsAsync<AccessList>();
+            return await getAccessListsResponse.Content.ReadAsAsync<AccessList>();
+        }
 
-            _listMembers.TryGetValue(identifier, out List<AccessListMemberDto> list);
+        public async Task<PagedAccessListMembersResponse> GetAccessListMembers(
+            string org,
+            string identifier,
+            string env,
+            string page
+        )
+        {
+            string listMembersUrl = string.IsNullOrEmpty(page) ? $"/{org}/{identifier}/members" : $"/{GetAccessListPageUrlSuffix(page, env)}";
+            HttpRequestMessage membersRequest = await CreateAccessListRequest(env, HttpMethod.Get, listMembersUrl);
 
-            IEnumerable<string> partyIds = (list ?? new List<AccessListMemberDto>()).Select(x => x.Identifiers.OrganizationNumber);
+            HttpResponseMessage geMembersResponse = await _httpClient.SendAsync(membersRequest);
+            geMembersResponse.EnsureSuccessStatusCode();
 
+            string membersResponseContent = await geMembersResponse.Content.ReadAsStringAsync();
+            AccessListMembersDto membersDto = JsonSerializer.Deserialize<AccessListMembersDto>(
+                membersResponseContent,
+                _serializerOptions
+            );
+
+            IEnumerable<string> partyIds = membersDto.Data.Select(x => x.Identifiers.OrganizationNumber);
+            List<AccessListMember> members = new List<AccessListMember>();
+
+            int BATCH_LOOKUP_SIZE = 190; // url cannot exceed 2048 characters, if there are many members, lookup names from brreg in batches
             // lookup party names
             if (partyIds.Any())
             {
-                string brregUrl = "https://data.brreg.no/enhetsregisteret/api/{0}?organisasjonsnummer={1}&size=10000";
-                string partyIdsString = string.Join(",", partyIds);
-                List<BrregParty>[] parties = await Task.WhenAll(
-                    GetBrregParties(string.Format(brregUrl, "enheter", partyIdsString)),
-                    GetBrregParties(string.Format(brregUrl, "underenheter", partyIdsString))
-                );
-
-                accessList.Members = partyIds.Select(orgnr =>
+                for (int i = 0; i < partyIds.Count(); i += BATCH_LOOKUP_SIZE)
                 {
-                    string enhetOrgName = parties[0].Find(enhet => enhet.Organisasjonsnummer.Equals(orgnr))?.Navn;
-                    string underenhetOrgName = parties[1].Find(enhet => enhet.Organisasjonsnummer.Equals(orgnr))?.Navn;
-                    AccessListMember member = new()
+                    IEnumerable<string> batchPartyIds = partyIds.Where((_x, index) => index >= i && index < (i + BATCH_LOOKUP_SIZE));
+                    string brregUrl = "https://data.brreg.no/enhetsregisteret/api/{0}?organisasjonsnummer={1}&size=10000";
```

Comment:
Later we should move this to some config

Reply:
Good point, made https://github.com/Altinn/altinn-studio/issues/12849 for this